src/structures/GuildEmoji.js

```diff
@@ -182,13 +120,13 @@ class GuildEmoji extends Emoji {
         other.name === this.name &&
         other.managed === this.managed &&
         other.requiresColons === this.requiresColons &&
-        other._roles === this._roles
+        other.roles.every(role => this.roles.has(role.id))
       );
     } else {
       return (
         other.id === this.id &&
         other.name === this.name &&
-        other._roles === this._roles
```

Comment:
In case of raw api data this would be an array of role ids and not role instances.